### PR TITLE
Refactor double conditional into one

### PR DESCRIPTION
**Pull request by Rodrigo-Novas**
Related to: https://github.com/urllib3/urllib3/issues/3117

### CODE REVIEW

**Comment by Rodrigo-Novas** [comment]
I don't know if this small change warrants a changelog entry.

**Comment by pquentin** [comment]
Well we could go one step further and write `flush_decoder = amt is None or (amt != 0 and not data)`. I'll let someone else weigh in as I wrote this code initially and am not sure if any of this makes a big difference.

**Comment by Rodrigo-Novas** [comment]
> Well we could go one step further and write `flush_decoder = amt is None or (amt != 0 and not data)`. I'll let someone else weigh in as I wrote this code initially and am not sure if any of this makes a big difference.

Yes good catch!. I think it is not a core change but having the code without **elifs** makes it more readable and without so much redundancy.

**Comment by pquentin** [comment]
Closing given the lack of interest, sorry.

File: src/urllib3/response.py [hunk 1]
```diff
@@ -878,11 +878,7 @@ def read(
 
         data = self._raw_read(amt)
 
-        flush_decoder = False
-        if amt is None:
-            flush_decoder = True
-        elif amt != 0 and not data:
-            flush_decoder = True
+        flush_decoder = True if amt is None or (amt != 0 and not data) else False
```
Comment (illia-v) [inline src/urllib3/response.py:881]: I like the change, `flush_decoder = amt is None or (amt != 0 and not data)` should be enough here

**Review by illia-v — APPROVED** [review]

**Pull request merged**